File: DevExpress-Demo/DevExpress.SalesDemo.Wpf/VB/DevExpress.SalesDemo.Wpf/ViewModel/MainViewModel/LoginViewModel.vb
Imports DevExpress.Mvvm.POCO
Imports DevExpress.SalesDemo.Wpf.Helpers
Imports System.Windows.Media.Imaging

Namespace DevExpress.SalesDemo.Wpf.ViewModel

    Public Class LoginViewModel

        Public Shared Function Create() As LoginViewModel
            Return ViewModelSource.Create(Function() New LoginViewModel())
        End Function

        Public Shared Function Clone(ByVal vm As LoginViewModel) As LoginViewModel
            Dim res = Create()
            res.UserName = vm.UserName
            Return res
        End Function

        Protected Sub New()
            Icon = ResourceImageHelper.GetResourceImage("User.png")
            UserName = "John Smith"
        End Sub

        Public Overridable Property Icon As BitmapImage

        Public Overridable Property UserName As String

        Public Overridable Property Password As String
    End Class
End Namespace

File: DevExpress-Demo/DevExpress.SalesDemo.Wpf/CS/DevExpress.SalesDemo.Wpf/ViewModel/MainViewModel/LoginViewModel.cs
using DevExpress.Mvvm.POCO;
using DevExpress.SalesDemo.Wpf.Helpers;
using System.Windows.Media.Imaging;

namespace DevExpress.SalesDemo.Wpf.ViewModel {
    public class LoginViewModel {
        public static LoginViewModel Create() {
            return ViewModelSource.Create(() => new LoginViewModel());
        }
        public static LoginViewModel Clone(LoginViewModel vm) {
            var res = Create();
            res.UserName = vm.UserName;
            return res;
        }
        protected LoginViewModel() {
            Icon = ResourceImageHelper.GetResourceImage("User.png");
            UserName = "John Smith";
        }

        public virtual BitmapImage Icon { get; protected set; }
        public virtual string UserName { get; set; }
        public virtual string Password { get; set; }
    }
}
